tests/unit-tests/dds/DCPS/XTypes/Utils.cpp

```diff
@@ -236,6 +236,46 @@ TEST_F(dds_DCPS_XTypes_Utils, get_keys)
   */
 }
 
+TEST_F(dds_DCPS_XTypes_Utils, member_path_resolve_string_path)
+{
+  add_type<AppendableMaxAppendableStruct>();
+  DDS::DynamicType_var dt = get_dynamic_type<AppendableMaxAppendableStruct>();
+
+  {
+    MemberPath path;
+    EXPECT_RC_OK(path.resolve_string_path(dt, "fmas.fs.value"));
+    EXPECT_EQ(path.level(), 3u);
+    EXPECT_EQ(path.ids[0], 0u);
+    EXPECT_EQ(path.ids[1], 0u);
+    EXPECT_EQ(path.ids[2], 0u);
+  }
+  {
+    MemberPath path;
+    EXPECT_RC_OK(path.resolve_string_path(dt, "fmas.au.value"));
+    EXPECT_EQ(path.level(), 3u);
+    EXPECT_EQ(path.ids[0], 0u);
+    EXPECT_EQ(path.ids[1], 1u);
+    EXPECT_EQ(path.ids[2], 0u);
+  }
+  {
+    MemberPath path;
+    EXPECT_RC_EQ(path.resolve_string_path(dt, "fmas."), DDS::RETCODE_BAD_PARAMETER);
+  }
+  {
+    MemberPath path;
+    EXPECT_NE(path.resolve_string_path(dt, "fmas.fs.invalid"), DDS::RETCODE_OK);
+  }
```

Comment:
Would be nicer if it can use the exact error with `EXPECT_RC_EQ`, although I wish this method returned `DDS::RETCODE_BAD_PARAMETER` in this case.

```suggestion
  {
    MemberPath path;
    EXPECT_RC_EQ(path.resolve_string_path(dt, "fmas.fs.invalid"), DDS::RETCODE_ERROR);
  }
```

Reply:
I didn't look into what error code can be returned by the `get_member_by_name` and `get_descriptor` calls inside this function. The only error code is `RETCODE_ERROR`, so this will work.